src/flamenco/runtime/fd_runtime.c

```diff
@@ -828,333 +778,225 @@ fd_runtime_execute_txn_task(void *tpool,
-          // FD_LOG_WARNING(("phase 2 invalid: %64J", (uchar *)txn_ctx->_txn_raw->raw+txn_ctx->txn_descriptor->signature_off ));
-          task_info[ txn_idx ].txn->flags = 0;
-          res |= FD_RUNTIME_TXN_ERR_BLOCKHASH_NOT_FOUND;
-          continue;
-        }
-      }
-
-      if( slot_ctx->status_cache ) {
-        fd_txncache_query_t curr_query;
-        curr_query.blockhash = blockhash->uc;
-        fd_blake3_t b3[1];
-        uchar hash[32];
-        fd_blake3_init( b3 );
-        fd_blake3_append( b3, ((uchar *)txn_ctx->_txn_raw->raw + txn_ctx->txn_descriptor->message_off),(ulong)( txn_ctx->_txn_raw->txn_sz - txn_ctx->txn_descriptor->message_off ) );
-        fd_blake3_fini( b3, hash );
-        curr_query.txnhash = hash;
-
-        // TODO: figure out if it is faster to batch query properly and loop all txns again
-        fd_txncache_query_batch( slot_ctx->status_cache, &curr_query, 1UL, query_arg, query_func, &err );
-
-        if( err != FD_RUNTIME_EXECUTE_SUCCESS ) {
-          // FD_LOG_WARNING(("phase 2 invalid: %64J", (uchar *)txn_ctx->_txn_raw->raw+txn_ctx->txn_descriptor->signature_off ));
-          task_info[ txn_idx ].txn->flags = 0;
-          res |= FD_RUNTIME_TXN_ERR_ALREADY_PROCESSED;
-          continue;
-        }
-      }
-
-      err = fd_executor_check_txn_accounts( txn_ctx );
-      if ( err != FD_RUNTIME_EXECUTE_SUCCESS ) {
-        // FD_LOG_WARNING(("phase 2 invalid: %64J", (uchar *)txn_ctx->_txn_raw->raw+txn_ctx->txn_descriptor->signature_off ));
-        task_info[ txn_idx ].txn->flags = 0;
-        res |= err;
-        continue;
-      }
-
-      fee_payer_idxs[fee_payer_accs_cnt] = txn_idx;
-      fee_payer_accs[fee_payer_accs_cnt] = fd_borrowed_account_init( &collect_fee_task_infos[fee_payer_accs_cnt].fee_payer_rec );
-      collect_fee_task_infos[fee_payer_accs_cnt].txn_ctx = txn_ctx;
-      collect_fee_task_infos[fee_payer_accs_cnt].result = !( task_info[txn_idx].txn->flags & FD_TXN_P_FLAGS_SANITIZE_SUCCESS ) ? -1 : 0;
-      fee_payer_accs_cnt++;
-    }
-
-    fd_tpool_exec_all_rrobin( tpool, 0, fd_tpool_worker_cnt( tpool ), fd_collect_fee_task, collect_fee_task_infos, NULL, NULL, 1, 0, fee_payer_accs_cnt );
+  FD_SCRATCH_SCOPE_BEGIN {
 
-    for (ulong fee_payer_idx = 0; fee_payer_idx < fee_payer_accs_cnt; fee_payer_idx++) {
-      fd_collect_fee_task_info_t * collect_fee_task_info = &collect_fee_task_infos[fee_payer_idx];
-      if( FD_UNLIKELY( collect_fee_task_info->result!=0 ) ) {
-        task_info[ fee_payer_idxs[ fee_payer_idx ] ].txn->flags = 0;
-        FD_LOG_WARNING(( "failed to collect fees" ));
-        res |= -1;
+    fd_tpool_exec_all_rrobin( tpool, 0, fd_tpool_worker_cnt( tpool ), fd_txn_pre_execute_checks_task, task_info, NULL, NULL, 1, 0, txn_cnt );
+    for( ulong txn_idx=0UL; txn_idx<txn_cnt; txn_idx++ ) {
+      if( FD_UNLIKELY( !( task_info[txn_idx].txn->flags & FD_TXN_P_FLAGS_SANITIZE_SUCCESS ) ) ) {
+        res |= FD_RUNTIME_TXN_ERR_ACCOUNT_LOADED_TWICE;
```

Comment:
@ibhatt-jumptrading `fd_txn_pre_execute_checks_task` can fail for multiple reasons (e.g., precompiles lol). Here, however, you're returning account loaded twice no matter what.